Improve "from" behavior with unnamed references

nalind opened the pull request:
Fix our instantiation behavior when the source image reference is not a named reference.  Take the opportunity to rename the `--registry` CLI flag to `--transport`.

rhatdan commented on new.go:
```diff
@@ -18,97 +21,113 @@ const (
 		}
 	}
 
+	name := "working-container"
```
Should this be made into a function?

rhatdan commented on docs/buildah-from.md:
```diff
@@ -46,13 +46,13 @@ If an image needs to be pulled from the registry, suppress progress output.
 ## EXAMPLE
 
-buildah from imagename --pull --registry "myregistry://"
+buildah from imagename --pull --transport "docker://myregistry.example.com/"
```
Why is this easier than 

buildah from docker://myregistry.example.com/imagename

nalind replied:
Both should provide equivalent results, but the `--transport` option is necessary for `build-using-dockerfile` anyway.

rhatdan replied:
Why can't I do 
buildah from docker://myregistry.example.com/imagename

nalind replied:
You can.

rhatdan replied:
If I can, why do I need --transport?

nalind commented:
Fixed an inconsistency (was `--default-transport` in `build-using-dockerfile`, instead of `--transport`, as it should have been), and added a note stressing that we don't insert any additional separator when combining the transport name with the image name.

nalind commented:
What errors are you getting back when you try to commit or push to a `docker-daemon:` destination?

rhatdan commented:
I am talking about https://github.com/projectatomic/buildah/issues/229

nalind commented:
Yes, this should fix those pull errors.

nalind commented:
Hmm, no it doesn't.

rhatdan commented:
Ok I will approve this patch and open another one to remove --transport.  Just so we can fix the issue.
@rh-atomic-bot r+ fb32f53
